docs/design/design-goals.md

```diff
@@ -0,0 +1,23 @@
+# General Design Goals
+
+Whether designing new features or redesigning old ones, OLM should maintain these general design goals:
+
+## Build (reasonably) small, sharp features
+
+All designs should be kept as simple and tightly scoped as is reasonably possible. Where possible, attempt to follow the [unix philosophy](https://en.wikipedia.org/wiki/Unix_philosophy) for writing small, sharp features that compose well. Additionally, when writing a new proposal, always remember to [keep it simple ...student!](https://en.wikipedia.org/wiki/KISS_principle)
```

Comment:
Just thinking out loud: Should we have a more formal process wrt design proposals here? What about https://github.com/operator-framework/enhancements ?

Reply:
I agree -- but I want to keep this doc more abstract. Could we add it in a separate `dev-process` doc?